docker/build_artifacts/sagemaker/python_service.py

```diff
@@ -10,54 +10,236 @@
+                # update model name <-> tfs pid map
+                self._model_tfs_pid[model_name] = p
+
+                res.status = falcon.HTTP_200
+                res.body = json.dumps({
+                    'success':
+                        'Successfully loaded model {}, '
+                        'listening on rest port {} '
+                        'and grpc port {}.'.format(model_name,
+                                                   self._model_tfs_rest_port,
+                                                   self._model_tfs_grpc_port,)
+                })
+            except MultiModelException as multi_model_exception:
+                if multi_model_exception.code == 409:
+                    res.status = falcon.HTTP_409
+                    res.body = multi_model_exception.msg
+                elif multi_model_exception.code == 408:
+                    res.status = falcon.HTTP_408
+                    res.body = multi_model_exception.msg
+                else:
+                    raise MultiModelException(falcon.HTTP_500, multi_model_exception.msg)
+            except FileExistsError as e:
+                res.status = falcon.HTTP_409
+                res.body = json.dumps({
+                    'error': 'Model {} is already loaded. {}'.format(model_name, str(e))
+                })
+            except OSError as os_error:
+                if os_error.errno == 12:
+                    raise MultiModelException(falcon.HTTP_507,
+                                              'Memory exhausted: '
+                                              'not enough memory to start TFS instance')
+                else:
+                    raise MultiModelException(falcon.HTTP_500, os_error.strerror)
+        else:
+            res.status = falcon.HTTP_404
+            res.body = json.dumps({
+                'error':
+                    'Could not find valid base path {} for servable {}'.format(base_path,
+                                                                               model_name)
+            })
+
+    def _wait_for_model(self, model_name):
+        url = "http://localhost:{}/v1/models/{}".format(self._model_tfs_rest_port[model_name],
+                                                        model_name)
+        with timeout():
+            while True:
+                time.sleep(0.5)
+                try:
+                    response = requests.get(url)
+                    if response.status_code == 200:
+                        versions = json.loads(response.content)['model_version_status']
+                        count = len(versions)
+                        for version in versions:
+                            state = version['state']
+                            if state == 'AVAILABLE':
+                                count -= 1
+                        if count == 0:
+                            break
+                except ConnectionError:
```

Comment:
log exception to provide some form of visibility even though the design is to swallow the exception?